Include/internal/pycore_floatobject.h

```diff
@@ -38,6 +38,9 @@ struct _Py_float_state {
 #endif
 };
 
+PyAPI_FUNC(void) _PyFloat_ExactDealloc(PyFloatObject *op);
```

Comment:
Drop the `PyAPI_FUNC`

Reply:
I tried, but mathmodule.c failed to compile on all but Windows. Should I keep it, or do I need to change some build configurations or something? I'm not that familiar with the whole build system.

Reply:
Symbols with leading underscores are not exported. So, modules can't access internal symbols.
Do you need to use the specialized forms in the math module?

Reply:
> Do you need to use the specialized forms in the math module?

I suppose not necessarily, but it has the same small positive effect in `math.prod` as it does in ceval.

Reply:
`math.prod` optimizations were not significant, so I removed them and the PyAPI_FUNC status